tests/test_FX_revaluation.py

```diff
@@ -0,0 +1,108 @@
+"""Unit tests for FX revaluation"""
+
+from io import StringIO
+from cashctrl_ledger import CashCtrlLedger
+import pandas as pd
+import pytest
+
+ACCOUNT_CSV = """
+    group, account, currency, vat_code, text
+    /Assets, 10001,      EUR,         , Test EUR Bank Account
+    /Assets, 10002,      USD,         , Test USD Bank Account
+    /Assets, 10003,      CHF,         , Test CHF Bank Account
+    /Assets, 10004,      USD,         , Test USD Bank Account 2
+"""
+
+# flake8: noqa: E501
+
+LEDGER_CSV = """
+    id,     date, account, counter_account, currency,     amount, base_currency_amount,      vat_code, text,                             document
+    1,  2024-05-24, 10001,           10003,      EUR,     100.00,                     ,              , pytest single transaction 1,
+    2,  2024-05-24, 10002,           10003,      USD,     100.00,                     ,              , pytest single transaction 2,
+    3,  2024-05-24, 10004,           10003,      USD,     100.00,                     ,              , pytest single transaction 3,
+"""
+
+# flake8: enable
+
+STRIPPED_CSV = "\n".join([line.strip() for line in LEDGER_CSV.split("\n")])
+LEDGER_ENTRIES = pd.read_csv(
+    StringIO(STRIPPED_CSV), skipinitialspace=True, comment="#", skip_blank_lines=True
+)
+TEST_ACCOUNTS = pd.read_csv(StringIO(ACCOUNT_CSV), skipinitialspace=True)
+
+
+@pytest.fixture(scope="session")
+def set_up_ledger_and_account():
+    cashctrl = CashCtrlLedger()
+    cashctrl.transitory_account = 1000
+
+    # Fetch original state
+    initial_account_chart = cashctrl.account_chart().reset_index()
+    initial_ledger = cashctrl.ledger()
+
+    # Create test accounts and Ledger
+    cashctrl.mirror_account_chart(TEST_ACCOUNTS, delete=False)
+    cashctrl.mirror_ledger(LEDGER_ENTRIES, delete=False)
+
+    yield
+
+    # Restore initial state
+    cashctrl.mirror_ledger(initial_ledger, delete=True)
+    cashctrl.mirror_account_chart(initial_account_chart, delete=True)
+
+
+def test_FX_revaluation(set_up_ledger_and_account):
+    cashctrl = CashCtrlLedger()
+    accounts = pd.DataFrame({
+        "foreign_currency_account": [10001, 10002, 10004],
+        "fx_gain_loss_account": [10003, 10003, 10003],
+        "exchange_rate": [0.75, 0.5, None],
+    })
+    eur_account_id = cashctrl._client.account_to_id(10001)
+    usd_account_id = cashctrl._client.account_to_id(10002)
+    usd_na_account_id = cashctrl._client.account_to_id(10004)
+
+    cashctrl.FX_revaluation(accounts=accounts)
+    ex_diff = cashctrl._client.get("fiscalperiod/exchangediff.json")["data"]
+    eur_account = next((item for item in ex_diff if item['accountId'] == eur_account_id), None)
+    usd_account = next((item for item in ex_diff if item['accountId'] == usd_account_id), None)
+    usd_na_account = next((item for item in ex_diff if item['accountId'] == usd_na_account_id), None)
```

Comment:
This seems overly complicated. I find the snippet hard to read. 

How about: Implementing `FX_valuation`, which calls `cashctrl._client.get("fiscalperiod/exchangediff.json")`, converts the result to a data frame a with an 'account' column instead of accountId? We could then drop accountId's in the test.